doc/api/packages.md

```diff
@@ -0,0 +1,830 @@
+# Modules: Packages
+
+<!-- type=misc -->
+
+## Introduction
+
+A package is a folder described by a `package.json` file.
+
+A folder containing a `package.json` file, and all subfolders below that folder
+until the next folder containing another `package.json` file, are a _package
+scope_.
+
```

Comment:
Can we add another sort of summary here as to what fields are defined here, and a pointer to npm definitions, maybe something like:

```
Node.js uses the following fields in the package.json:

* [`"type"`](#link)
* [`"main"`](#link)
* [`"exports"`](#link)
* [`"imports"`](#link)

Other projects define and use their own fields, such as those used by [npm for defining dependencies](https://docs.npmjs.com/files/package.json).
```

Reply:
I'd very much prefer to do that in a following PR if that's OK: I promise you that's something I want to do (it's been several months I have started pushing for it 😅), I'm worried nothing will get done if I try to do it in one PR.

Reply:
Ok great, let's do it next of course.